addons/mail/wizard/mail_compose_message.py

```diff
@@ -819,3 +851,45 @@ def _generate_template_for_composer(self, template, res_ids, render_fields,
             }
 
         return render_results
+
+    # ----------------------------------------------------------------------
+    # MISC UTILS
+    # ----------------------------------------------------------------------
+
+    @api.model
+    def _parse_res_domain(self, domain_str):
```

Comment:
we can even go futher in the check if we give the model, if we delegate the verification to expression.py (it will be simpler, and check more so it's win/win)

something like this

```python
try:
    if isinstance(domain, str):
        domain = ast.literal_eval(domain)

    expression.expression(
        domain,
        self.env[res_model],
    )
except ValueError, AssertionError:
    raise ValidationError(
        _("Invalid domain %(domain)r (type %(domain_type)s)",
            domain=domain,
            domain_type=type(domain_str)))

return domain
```

The only downside is that `_parse_res_domain` must either receive the model name, or be called on a singleton

What do you think ?

Reply:
Good idea :)  will give it a try